packages/front-end/services/datasources.ts

```diff
@@ -939,44 +1022,64 @@ SELECT
   (_TABLE_SUFFIX BETWEEN 'intraday_{{date startDateISO "yyyyMMdd"}}' AND 'intraday_{{date endDateISO "yyyyMMdd"}}'))
             `.trim(),
-            eventName: "",
-            userIdTypes,
-            columns: generateColumns({
-              timestamp: { datatype: "date" },
-              user_id: { datatype: "string" },
-              anonymous_id: { datatype: "string" },
-              country: { datatype: "string" },
-              device_category: { datatype: "string" },
-              source: { datatype: "string" },
-              medium: { datatype: "string" },
-              campaign: { datatype: "string" },
-              page_path: { datatype: "string", alwaysInlineFilter: true },
-              session_id: { datatype: "string" },
-            }),
-          },
-          filters: [
-            {
-              name: "Desktop",
-              description: "Events fired on desktop devices",
-              value: `device_category = 'desktop'`,
-            },
-            {
-              name: "Mobile / Tablet",
-              description: "Events fired on mobile or tablet devices",
-              value: `device_category IN ('mobile', 'tablet')`,
-            },
-          ],
-          metrics: [],
+          eventName: "",
+          userIdTypes,
+          columns: generateColumns({
+            timestamp: { datatype: "date" },
+            user_id: { datatype: "string" },
+            anonymous_id: { datatype: "string" },
+            country: { datatype: "string" },
+            device_category: { datatype: "string" },
+            source: { datatype: "string" },
+            medium: { datatype: "string" },
+            campaign: { datatype: "string" },
+            page_path: { datatype: "string", alwaysInlineFilter: true },
+            session_id: { datatype: "string" },
+          }),
         },
-      ],
-    };
-  }
+        filters: [
+          {
+            name: "Desktop",
+            description: "Events fired on desktop devices",
+            value: `device_category = 'desktop'`,
+          },
+          {
+            name: "Mobile / Tablet",
+            description: "Events fired on mobile or tablet devices",
+            value: `device_category IN ('mobile', 'tablet')`,
+          },
+        ],
+        metrics: [],
+      },
+    ],
+  };
+};
+
+export function getInitialDatasourceResources({
+  datasource,
+}: {
+  datasource: DataSourceInterfaceWithParams;
+}): InitialDatasourceResources {
+  if (
+    datasource.type === "bigquery" &&
+    datasource.settings?.schemaFormat === "ga4"
+  )
+    return getBigQueryWithGa4InitialDatasourceResources({
+      params: datasource.params,
+      userIdTypes: datasource.settings?.userIdTypes || [],
+    });
+
+  if (datasource.type === "growthbook_clickhouse")
+    return getClickHouseInitialDatasourceResources({
+      table: process.env.CLICKHOUSE_MAIN_TABLE || "test_enriched_events",
```

Comment:
You can't access `process.env` on the front-end.  Better if we can lookup the table from the `datasource.settings` or `datasource.params` object instead.